cmd/klotho/main.go

```diff
@@ -264,13 +261,15 @@ func run(cmd *cobra.Command, args []string) (err error) {
 	analyticsClient.Info("klotho compiling")
 
 	result, err := compiler.Compile(input)
-	if err != nil {
-		errHandler.PrintErr(err)
+	if err != nil || hadErrors.Load() {
+		if err != nil {
+			errHandler.PrintErr(err)
+		}
 		analyticsClient.Error("klotho compiling failed")
 
 		cmd.SilenceErrors = true
 		cmd.SilenceUsage = true
-		return err
+		return errors.New("Failed run of klotho invocation")
```

Comment:
because we always set SilenceErrors to true it shouldnt matter what we return, just that we return some error